sharding/contracts/sharding_manager.sol

```diff
@@ -69,9 +70,6 @@ contract SMC {
     // Threshold(number of notaries in committee) for a proposal to be accepted
     uint constant QUORUM_SIZE = 90;
 
-    // Log the latest period number of the shard
-    mapping (int => int) public periodHead;
-
     /// Checks if a notary with given shard id and period has been chosen as
     /// a committee member to vote for header added on to the main chain
     function getNotaryInCommittee(uint shardId, uint _index) public view returns(address) {
```

Comment:
We need to check `shardId` against `SHARD_COUNT` either in here or in `submit_vote`(or both).

Reply:
Good call. Added check for `shardId` in `submit_vote`